papis/config.py

```diff
@@ -188,61 +208,56 @@ def register_default_settings(settings_dictionary: PapisConfigType) -> None:
             default_settings[section] = settings
 
 
-def get_config_home() -> str:
-    """
-    :returns: the base directory relative to which user specific configuration
-        files should be stored.
-    """
-    xdg_home = os.environ.get("XDG_CONFIG_HOME")
-    if xdg_home:
-        return os.path.expanduser(xdg_home)
+def _deprecated_config_folder() -> str:
+    xdg_config_home = os.environ.get("XDG_CONFIG_HOME")
+    if xdg_config_home:
+        xdg_config_home = os.path.expanduser(xdg_config_home)
     else:
-        return os.path.join(os.path.expanduser("~"), ".config")
+        xdg_config_home = os.path.join(os.path.expanduser("~"), ".config")
 
-
-def get_config_dirs() -> List[str]:
-    """
-    :returns: a list of directories where the configuration files might be stored.
-    """
-    dirs: List[str] = []
+    config_dirs: List[str] = []
 
     # get_config_home should also be included on top of XDG_CONFIG_DIRS
-    config_dirs = os.environ.get("XDG_CONFIG_DIRS")
-    if config_dirs:
-        dirs += [os.path.join(d, "papis") for d in config_dirs.split(":")]
+    xdg_config_dirs = os.environ.get("XDG_CONFIG_DIRS")
+    if xdg_config_dirs:
+        config_dirs += [os.path.join(d, "papis") for d in xdg_config_dirs.split(":")]
 
     # NOTE: Take XDG_CONFIG_HOME and $HOME/.papis for backwards compatibility
-    dirs += [
-        os.path.join(get_config_home(), "papis"),
+    config_dirs += [
+        os.path.join(xdg_config_home, "papis"),
         os.path.join(os.path.expanduser("~"), ".papis")]
 
-    return dirs
+    for config_dir in config_dirs:
+        if os.path.exists(config_dir):
+            return config_dir
 
+    # NOTE: If no folder is found, then get the config home
+    return os.path.join(xdg_config_home, "papis")
 
-def get_config_folder() -> str:
-    """Get the main configuration folder.
 
-    :returns: the folder where the configuration files are stored, e.g.
-        ``$HOME/.config/papis``, by looking in the directories returned by
-        :func:`get_config_dirs`.
+def get_config_home() -> str:
     """
-    config_dirs = get_config_dirs()
+    :returns: a (platform dependent) base directory relative to which user
+        specific configuration files should be stored.
+    """
+    return platformdirs.user_config_dir()
 
-    for config_dir in config_dirs:
-        if os.path.exists(config_dir):
-            return config_dir
 
-    # NOTE: If no folder is found, then get the config home
-    return os.path.join(get_config_home(), "papis")
+def get_config_folder() -> str:
+    """Get the main configuration folder.
+
+    :returns: a (platform dependent) folder where the configuration files are
+        stored, e.g. ``$HOME/.config/papis`` on POSIX platforms.
+    """
+    return platformdirs.user_config_dir("papis")
```

Comment:
This doesn't look through `XDG_CONFIG_DIRS` anymore, so it won't pick up configurations in e.g. `/etc/xdg` on Linux. It should be easy enough to add if someone needs it though.